src/runtime_src/core/pcie/driver/linux/xocl/subdev/qdma.c

```diff
@@ -1695,15 +1696,26 @@ static int qdma_probe(struct platform_device *pdev)
 	qdma->pdev = pdev;
 	platform_set_drvdata(pdev, qdma);
 
-	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-	if (!res) {
-		xocl_err(&pdev->dev, "Empty resource");
-		return -EINVAL;
+	for (i = 0, res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
+		res;	
+		res = platform_get_resource(pdev, IORESOURCE_MEM, ++i)) {
+		if (!strncmp(res->name, NODE_QDMA, strlen(NODE_QDMA))) {
+			ret = xocl_ioaddr_to_baroff(xdev, res->start, &dma_bar, NULL);
+			if (ret) {
+				xocl_err(&pdev->dev, "Invalid resource %pR", res);
+				return -EINVAL;
+			}
+		} else if (!strncmp(res->name, NODE_STM, strlen(NODE_STM))) {
+			stm_base = res->start -
+				pci_resource_start(XDEV(xdev)->pdev, XDEV(xdev)->bar_idx);
+		} else {
+			xocl_err(&pdev->dev, "Unknown resource: %s", res->name);
+			return -EINVAL;
+		}
 	}
 
-	ret = xocl_ioaddr_to_baroff(xdev, res->start, &dma_bar, NULL);
-	if (ret) {
-		xocl_err(&pdev->dev, "Invalid resource %pR", res);
+	if (dma_bar == -1 || stm_base == -1) {
```

Comment:
legacy QDMA device does not have stm defined. Please add STM to XOCL_RES_QDMA. Please verify legacy QDMA platform.

Reply:
will change